File: algorithms_toolbox/week3_greedy_algorithms/1_money_change/Change.java
import java.util.Scanner;

public class Change {
    private static int getChange(int m) {
        //write your code here
    	int numCoins = 0;
    	int[] d = {10, 5, 1};
    	for(int i = 0; i < d.length; i++) {
    		if(m > 0) {
    			numCoins += m / d[i];
    			m %= d[i];
    		}else {break;}
    	}
        return numCoins;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int m = scanner.nextInt();
        System.out.println(getChange(m));

    }
}

File: algorithms_toolbox/week3_greedy_algorithms/1_money_change/change.py
# Uses python3
import sys

def get_change(m):
    coins = [1, 5, 10]
    num_coins = 0
    #write your code here
    for coin in coins[::-1]:
        num_coins += int(m / coin)
        m %= coin
    return num_coins

if __name__ == '__main__':
    m = int(sys.stdin.readline())
    print(get_change(m))
